Automattic/vip-go-mu-plugins
# Schedule multiple processor jobs on shorter interval

## Description

The sweeper needs to run more often and also process more jobs per run so that when a massive amount of indexing operations are offloaded to be processed asynchronously, they are processed in a timely manner.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

Before doing any testing, set `PROCESSOR_MAX_OBJECTS_PER_CRON_EVENT` to 10

---

1. Ensure cron-control isn't active.
2. Apply PR
3. Ensure you have lots of posts in your test instance and `wp vip-search queue stress test` to populate `wp_vip_search_index_queue`.
2. `wp shell` and run `Automattic\VIP\Search\Search::instance()->queue->cron->sweep_jobs()`
3. Only one `vip_search_queue_processor` event should be scheduled(can be checked via `wp option get cron` or other method).

-----

1. Ensure cron-control is active and running properly.
2. Apply PR
3. Ensure you have lots of posts in your test instance and `wp vip-search queue stress test` to populate `wp_vip_search_index_queue`
3. Only five `vip_search_queue_processor` events should be scheduled(can be checked via `SELECT COUNT(*) FROM wp_a8c_cron_control_jobs WHERE action = 'vip_search_queue_processor' AND status != 'complete';` if you're fast enough, otherwise fine the general timestamp window in the table and compare).
4. Repeat 2 a few times and check again. Should still only be five events at any time.

-----

1. Ensure cron-control is active and running properly.
2. Apply PR.
3. Ensure `vip_search_queue_sweeper_interval` is now 1 minute instead of 5.
